types/vision/v4/index.d.ts

```diff
@@ -0,0 +1,212 @@
+    layoutKeyword?: string;
+    /** the text encoding used by the templates when reading the files and outputting the result. Defaults to 'utf8'. */
+    encoding?: string;
+    /** if set to true, allows absolute template paths passed to reply.view(). Defaults to false. */
+    allowAbsolutePaths?: boolean;
+    /** if set to true, allows template paths passed to reply.view() to contain '../'. Defaults to false. */
+    allowInsecureAccess?: boolean;
+    /** options object passed to the engine's compile function. Defaults to empty options {}. */
+    compileOptions?: CompileOptions;
+    /** options object passed to the returned function from the compile operation. Defaults to empty options {}. */
+    runtimeOptions?: RuntimeOptions;
+    /** the content type of the engine results. Defaults to 'text/html'. */
+    contentType?: string;
+    /** specify whether the engine compile() method is 'sync' or 'async'. Defaults to 'sync'. */
+    compileMode?: 'sync' | 'async';
+    /** a global context used with all templates. The global context option can be either an object or a function that takes the request as its only argument and returns a context object. The request object is only provided when using the view handler or reply.view(). When using server.render() or request.render(), the request argument will be null. When rendering views, the global context will be merged with any context object specified on the handler or using reply.view(). When multiple context objects are used, values from the global context always have lowest precedence. */
+    context?: Object | ((request: Hapi.Request) => Object);
+}
+
+/**
+ * Options passed to module when compiling template.
+ * Cast your options to this interface or extend it with:
+ *      declare module 'hapi' {
+ *          interface CompileOptions {
+ *              noEscape: boolean;
+ *          }
+ *      }
+ */
+export interface CompileOptions {}
+export interface RuntimeOptions {}
+
+/**
+ * the rendering function. The required function signature depends on the compileMode settings (see below).
+ * If compileMode is 'sync', the signature is compile(template, options), the return value is a function with signature function(context, options) (the compiled sync template), and the method is allowed to throw errors.
+ * If compileMode is 'async', the signature is compile(template, options, next) where next has the signature function(err, compiled), compiled is a function with signature function(context, options, callback) (the compiled async template) and callback has the signature function(err, rendered).
+ */
+export interface ServerViewCompile {
+    (template: string, options: any): (context: any, options: any) => void;
+    (template: string, options: any, next: (err: Error | null, compiled: (context: any, options: any, callback: (err: null | Error, rendered: string | null) => void) => void) => void): void;
```

Comment:
This type definition doesn't look quite right, triple nested callback?
If it is triple nested (I hope it's not), try to declare some intermediate types.